geiger/geiger.py

```diff
@@ -17,41 +17,53 @@
-submission = pd.read_csv('../input/jigsaw-toxic-comment-classification-challenge/sample_submission.csv')
-
-X_train = train["comment_text"].fillna("fillna").values
-y_train = train[["toxic", "severe_toxic", "obscene", "threat", "insult", "identity_hate"]].values
-X_test = test["comment_text"].fillna("fillna").values
-
-
-max_features = 30000
-maxlen = 100
-embed_size = 300
-
-tokenizer = text.Tokenizer(num_words=max_features)
-tokenizer.fit_on_texts(list(X_train) + list(X_test))
-X_train = tokenizer.texts_to_sequences(X_train)
-X_test = tokenizer.texts_to_sequences(X_test)
-x_train = sequence.pad_sequences(X_train, maxlen=maxlen)
-x_test = sequence.pad_sequences(X_test, maxlen=maxlen)
-
-
-def get_coefs(word, *arr): return word, np.asarray(arr, dtype='float32')
-embeddings_index = dict(get_coefs(*o.rstrip().rsplit(' ')) for o in open(EMBEDDING_FILE))
-
-word_index = tokenizer.word_index
-nb_words = min(max_features, len(word_index))
-embedding_matrix = np.zeros((nb_words, embed_size))
-for word, i in word_index.items():
-    if i >= max_features: continue
-    embedding_vector = embeddings_index.get(word)
-    if embedding_vector is not None: embedding_matrix[i] = embedding_vector
-
+import argparse
+
+parser = argparse.ArgumentParser(description='The geiger programm')
+parser.add_argument('train', type=str, help='The csv with the input data')
+parser.add_argument('test', type=str, help='The csv with the test data')
+parser.add_argument('embeddings', type=str, help='The embeddings file (e.g. fasttext)')
+parser.add_argument('--features', type=int, default=30000, help='The numbers of features of the embedding file. (in the first line of the file)')
+parser.add_argument('--maxlen', type=int, default=100, help='The length of the max sequence')
+parser.add_argument('--embed_size', type=int, default=300, help='The size of the embedding layer')
+parser.add_argument('--batch_size', type=int, default=32, help='Batch size ')
+parser.add_argument('--epochs', type=int, default=2, help='Total number of epochs to run for')
+
+args = parser.parse_args()
+
+
+# EMBEDDING_FILE = '../input/fasttext-crawl-300d-2m/crawl-300d-2M.vec'
+
+def load_files():
+    train = pd.read_csv(args.train)
+    test = pd.read_csv(args.test)
+
+    X_train = train["comment_text"].fillna("fillna").values
+    y_train = train[["toxic", "severe_toxic", "obscene", "threat", "insult", "identity_hate"]].values
+    X_test = test["comment_text"].fillna("fillna").values
+
+    tokenizer = text.Tokenizer(num_words=args.features)
+    tokenizer.fit_on_texts(list(X_train) + list(X_test))
+    X_train = tokenizer.texts_to_sequences(X_train)
+    X_test = tokenizer.texts_to_sequences(X_test)
+    x_train = sequence.pad_sequences(X_train, maxlen=args.maxlen)
+    x_test = sequence.pad_sequences(X_test, maxlen=args.maxlen)
+    return x_train, x_test, y_train, tokenizer
+
+def get_coefs(word, *arr):
+    return word, np.asarray(arr, dtype='float32')
+
+def load_embeddings(tokenizer):
```

Comment:
I wonder if it would be better to just use gensim to load the embeddings. For example, we could [this method](https://radimrehurek.com/gensim/models/fasttext.html#gensim.models.fasttext.FastText.load_fasttext_format) to load the original embeddings and have the model to abstract the logic for accessing word and char vectors.